### DISCUSSION TITLE
how to fully custom routes+names without nuxt convention?

### DISCUSSION THREAD

**Discussion by mhf-ir** [nuxt/nuxt#19386] [Questions]
How to set full customize routeing system and bypass nuxt folder base convention? also i need `named` base route system with custom routing names.
```js
[{
  path: '/CustomName',
  component: '~/folder1/CustomName.vue',
  name: 'notConvetionNeeded',
  children: [{
    path: 'child',
    component: '~/folder2/CustomNameSample2.vue',
    name: 'alsoNotConvetionNeeded',
    meta: {
      key: value
    }
  }]
}]
```

**Comment by Baroshem** [comment]
Hey,

You should be able to configure the router from nuxt configuration like following

https://nuxt.com/docs/api/configuration/nuxt-config/#router

**Reply by mhf-ir** [reply]
> Hey,
> 
> You should be able to configure the router from nuxt configuration like following
> 
> https://nuxt.com/docs/api/configuration/nuxt-config/#router

Is this will be completely disable auto generation nuxt routes convention ?
